slice/validate.go

```diff
@@ -35,5 +30,55 @@ func (s *Split) init() error {
 		return err
 	}
 
+	if err := s.validateFilters(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Split) validateFilters() error {
+	if len(s.opts.IncludedKinds) > 0 && len(s.opts.ExcludedKinds) > 0 {
+		return fmt.Errorf("cannot specify both included and excluded kinds")
+	}
+
+	if len(s.opts.IncludedNames) > 0 && len(s.opts.ExcludedNames) > 0 {
+		return fmt.Errorf("cannot specify both included and excluded names")
+	}
+
+	if len(s.opts.Included) > 0 && len(s.opts.Excluded) > 0 {
+		return fmt.Errorf("cannot specify both included and excluded")
+	}
+
+	// Merge all filters into excluded and included.
+	for _, v := range s.opts.IncludedKinds {
+		s.opts.Included = append(s.opts.Included, fmt.Sprintf("%s/*", v))
+	}
+
+	for _, v := range s.opts.ExcludedKinds {
+		s.opts.Excluded = append(s.opts.Excluded, fmt.Sprintf("%s/*", v))
+	}
+
+	for _, v := range s.opts.IncludedNames {
+		s.opts.Included = append(s.opts.Included, fmt.Sprintf("*/%s", v))
+	}
+
+	for _, v := range s.opts.ExcludedNames {
+		s.opts.Excluded = append(s.opts.Excluded, fmt.Sprintf("*/%s", v))
+	}
+
+	// Validate included and excluded filters.
+	for _, included := range s.opts.Included {
+		if matched, _ := regexp.MatchString(regKN, included); !matched {
+			return fmt.Errorf("invalidad included pattern '%s' should be {kind}/${name}", included)
```

Comment:
Same as above, the pattern can simply be `<kind>/<name>` without dollar signs and curly braces. Otherwise it might give the impression of an environment variable for interpolation.

There's also a typo:

```suggestion
			return fmt.Errorf("invalid included pattern %q should be <kind>/<name>", included)
```